### DISCUSSION TITLE
What is the action mapping when using the full legal action set?

### DISCUSSION THREAD

**Discussion by djbyrne** [sail-sg/envpool#263] [Q&A]
I am curious as to how envpool handles actions when using the full action set that are outside the minimal action set for the environment. 

For example if a game only uses 8/18 actions, and an action outside of the 8 is given, what does the environment do? 
1) is it a No-Op?
2) is the previous action repeated?
3) is there specific mappings from multiple actions to a single action (i.e are all the UPFIRE, LEFTFIRE, DOWNFIRE just mapped to FIRE) ? 

If anyone can point me to where this is defined I would greatly appreciate it!

**Comment by Trinkle23897** [comment]
https://github.com/sail-sg/envpool/blob/51f468604fd0f21ae19a970e88f893433dc4b4ec/envpool/atari/atari_env.h#L191
action will be mapped from action_set_, where it's from https://github.com/sail-sg/envpool/blob/51f468604fd0f21ae19a970e88f893433dc4b4ec/envpool/atari/atari_env.h#L129-L133

It will use ale's interface, same as https://github.com/mgbellemare/Arcade-Learning-Environment/blob/259f24951d27bdfcb5d7b3f54f1f420ca44b71ef/src/python/env/gym.py#L256

https://github.com/mgbellemare/Arcade-Learning-Environment/blob/259f24951d27bdfcb5d7b3f54f1f420ca44b71ef/src/python/ale_python_interface.hpp#L47-L49
